courses/views.py

```diff
@@ -53,4 +60,27 @@ def assignment_detail(request, assignment_id):
         raise Http404("Assignment does not exist")
     return render(request, 'assignments/detail.html', {'assignment': assignment })
 
+def add_course(request):
+    #TODO: Ensure only logged-in admins can get here
+    #Can we write a decorator to do the required test?
+    try:
+        userobj = request.user.user
+    except (User.DoesNotExist, AttributeError):
+        errormsg = "Cannot add courses: not logged in"
+    else: 
+        try:
+            if userobj.is_admin():
+                coursename = request.POST['course_name']
+                new_course = Course(name = coursename)
+                new_course.save()
+                return HttpResponseRedirect(reverse('courses:detail', args=[new_course.id]))
+            else:
+                errormsg = "Must have admin access to add courses"
+        except KeyError:
+            errormsg = "Invalid request to add_course"
+        except DataError:
+            errormsg = "Invalid course name"
```

Comment:
DataError is thrown when an overly long string is used, in this case. I would expect IntegrityError might come up, since it has to do with foreign keys and other such constraints.